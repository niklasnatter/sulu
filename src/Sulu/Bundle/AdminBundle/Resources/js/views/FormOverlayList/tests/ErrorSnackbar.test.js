// @flow
import React from 'react';
import {render, shallow} from 'enzyme';
import ErrorSnackbar from '../ErrorSnackbar';

jest.mock('../../../utils/Translator', () => ({
    translate: jest.fn((key) => key),
}));

test('Render an invisible error-snackbar', () => {
    expect(render(<ErrorSnackbar visible={false} />)).toMatchSnapshot();
});

test('Render an error-snackbar', () => {
    expect(render(<ErrorSnackbar onCloseClick={jest.fn()} />)).toMatchSnapshot();
});

test('Render an error-snackbar without close button', () => {
    expect(render(<ErrorSnackbar />)).toMatchSnapshot();
});

test('Call onCloseClick callback when close button is clicked', () => {
    const closeClickSpy = jest.fn();
    const snackbar = shallow(<ErrorSnackbar onCloseClick={closeClickSpy} />);

    snackbar.find('button').simulate('click');

    expect(closeClickSpy).toBeCalledWith();
});
