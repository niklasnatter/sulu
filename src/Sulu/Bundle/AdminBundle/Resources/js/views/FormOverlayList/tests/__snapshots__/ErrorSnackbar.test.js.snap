// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render an error-snackbar 1`] = `
<div
  class="snackbar visible"
>
  <span
    aria-label="su-exclamation-triangle"
    class="su-exclamation-triangle icon"
  />
  <div
    class="text"
  >
    <strong>
      sulu_admin.error
    </strong>
    <button
      class="closeButton"
    >
      sulu_admin.close
      <span
        aria-label="su-times"
        class="su-times closeButtonIcon"
      />
    </button>
  </div>
</div>
`;

exports[`Render an error-snackbar without close button 1`] = `
<div
  class="snackbar visible"
>
  <span
    aria-label="su-exclamation-triangle"
    class="su-exclamation-triangle icon"
  />
  <div
    class="text"
  >
    <strong>
      sulu_admin.error
    </strong>
  </div>
</div>
`;

exports[`Render an invisible error-snackbar 1`] = `
<div
  class="snackbar"
>
  <span
    aria-label="su-exclamation-triangle"
    class="su-exclamation-triangle icon"
  />
  <div
    class="text"
  >
    <strong>
      sulu_admin.error
    </strong>
  </div>
</div>
`;
